### PR TITLE
proposal of an update to your PR

### PR DESCRIPTION
**Pull request by sensorario**
First of all thank you for your pull request and also for this:

    // Best effort attempt, don't want to interfere with the logic
    // below, I'll let the maintainer decide how to use this
    // helper.

This could be a way to use your helper.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -22,26 +22,33 @@ func genLog() gol.Logger {
...
-	if repo, err := findRepository(dir); err == nil {
-		_ = os.Chdir(repo)
+
+	repositoryRoot, err := findRepository(currentFolder)
+
+	if err == nil {
+		_ = os.Chdir(repositoryRoot)
+	}
+
+	repositoryExists := true
```
Comment (nicolagi) [inline main.go]: This is more or less how I'd think you'd use the helper :)

One comment only, it doesn't feel right to check against "" to tell whether we found a repository or not. Would it be okay for you to define `repositoryExists := !os.IsNotExist(err)`?

File: main.go [hunk 2]
```diff
@@ -57,14 +64,13 @@ func main() {
 				// @todo ask what is the development branche
 
 				confIndented, _ := json.MarshalIndent(conf, "", "  ")
-				if _, err := os.Stat(dir + "/.git/ff.conf.json"); os.IsNotExist(err) {
-					_ = ioutil.WriteFile(dir+"/.git/ff.conf.json", confIndented, 0644)
+				if _, err := os.Stat(repositoryRoot + "/.git/ff.conf.json"); os.IsNotExist(err) {
```
Comment (nicolagi) [inline main.go]: Why not get rid of `dir+` altogether? If at the beginning of the program you chdir to the right dir, you could use relative paths from then on.

File: main.go [hunk 3]
```diff
@@ -22,26 +22,33 @@ func genLog() gol.Logger {
...
+
+	repositoryExists := true
+	if repositoryRoot == "" {
+		repositoryExists = false
+		fmt.Println(color.RedString("No repository found"))
 	}
 
-	conf := ReadConfiguration()
-	dir, _ = os.Getwd()
+	conf, err := ReadConfiguration(repositoryRoot)
```
Comment (nicolagi) [inline main.go]: Shouldn't we wait to read the configuration, until after we've established there is a repository or one was created with the workflow below?

Perhaps right after find repository you could have something like `if (os.IsNotExist(err)) { guidedRepositoryCreation() }` which encapsulates the below?

File: main.go (hunk 1 above)
Comment (sensorario) [reply main.go]: Great! Feel free to merge my suggestions or to complete your PR ^_^. I really appreciate your contribute.

File: main.go (hunk 2 above)
Comment (sensorario) [reply main.go]: Correct!

File: main.go (hunk 3 above)
Comment (sensorario) [reply main.go]: Good point. Absolutely yess!

**Pull request merged**